wlauto/utils/fps.py

```diff
@@ -0,0 +1,144 @@
+#    Copyright 2016 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import logging
+
+logger = logging.getLogger('fps-processor')
+
+
+class FpsProcessor(object):
+    """
+    Provide common object for processing surfaceFlinger output for frame
+    statistics.
+
+    This processor adds four metrics to the results:
+
+        :FPS: Frames Per Second. This is the frame rate of the workload.
+        :frames: The total number of frames rendered during the execution of
+                 the workload.
+        :janks: The number of "janks" that occurred during execution of the
+                workload. Janks are sudden shifts in frame rate. They result
+                in a "stuttery" UI. See http://jankfree.org/jank-busters-io
+        :not_at_vsync: The number of frames that did not render in a single
+                       vsync cycle.
+    """
+
+    def __init__(self, context, data, generate_csv=True, action=None):
+        """
+        context      - the wa execution context
+        data         - a pandas.DataFrame object with frame data (e.g. frames.csv)
+        generate_csv - if true create a csv file containing the metrics
+        action       - output metrics names with additional action specifier
+        """
+        self.context = context
```

Comment:
Utils modules are intended to be stand-alone bits of re-usable functionality. They cannot rely on the rest of WA framework. This should not be ware of "context" or "metrics" . Rather than adding the metrics to the contest's results, this should be returning the the calculated results and the jtg